### DISCUSSION TITLE
Validating auth in SSR context

### DISCUSSION THREAD

**Discussion by genu** [directus/directus#15068] [Archive]
Hi, I'm working on a nuxt3 project and trying to see if I can use directus as a backend.

A challange has been synchronizign the auth information between client and SSR contexts.

### The setup

When a client logs in, I do:
```ts
const authResult = await $directus.auth.login({ email, password })

useCookie('auth_token').value = authResult.access_token
useCookie<number>('auth_expires').value = authResult.expires
```

The [useCookie](https://v3.nuxtjs.org/api/composables/use-cookie) here just makes SSR friendly cookies copies of token and expiration.

When the page loads initially, in the SSR context, I initialize directus like this:
```ts
if (process.server) {
 // Get these values from the cookies that were created on client side.
  const auth_token = useCookie('auth_token').value
  const auth_expires = useCookie('auth_expires').value as unknown as number

  if (auth_token) {
    directus.storage.auth_token = auth_token
  }

  if (auth_expires) {
    directus.storage.auth_expires = auth_expires
  }
}
```

### The problem
In the above setup, the client logs in and sets cookies, and the SSR the reads the cookie. 
The issue i'm facing:

When a page loads, how can I check that an `auth_token` is valid? This way, I can redirect user to a login page. Is there a method to call in the js SDK to verify a token?

**Comment by rijkvanzanten** [comment]
> When a page loads, how can I check that an auth_token is valid? This way, I can redirect user to a login page. Is there a method to call in the js SDK to verify a token?

The quickest way to definitely know whether an auth session is valid is to check `await directus.users.me.read();` If there's no current auth session, there's no current user.

**Reply by HadiSDev** [reply]
Where do I find this in the docs? I cant find that function.

**Reply by br41nslug** [reply]
https://docs.directus.io/reference/system/users.html#retrieve-the-current-user

It doesnt have to be this endpoint tho, the trick here is to just do any authenticated request and if it fails the token isnt valid else it is.
